pkg/router/template/plugin.go

```diff
@@ -253,11 +254,41 @@ func peerEndpointsKey(namespacedName ktypes.NamespacedName) string {
 }
 
 // createRouterEndpoints creates openshift router endpoints based on k8s endpoints
-func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool) []Endpoint {
+func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, service *kapi.Service) []Endpoint {
+	// check if this service is currently idled
+	subsets := endpoints.Subsets
+	if _, ok := endpoints.Annotations[unidlingapi.IdledAtAnnotation]; ok && len(endpoints.Subsets) == 0 {
+		if service == nil {
+			// TODO: actually return an error
+			glog.Errorf("no idled service found corresponding to idled endpoints %v/%v", endpoints.Namespace, endpoints.Name)
+			return []Endpoint{}
+		}
+
+		svcSubset := kapi.EndpointSubset{
```

Comment:
Yeah, there's not a good way to do unidling with headless services.  I'll make sure we handle that case gracefully.